utils/sendInStakingWindow.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const RETRY_INTERVAL_MS = 2499;
+
+async function getCurrentBlockNumber(web3) {
+    return parseInt((await web3.eth.getBlock('latest')).number);
+}
+
+module.exports = async function (web3, sendTx) {
+    let ValidatorSetContract = require('../utils/getContract')('ValidatorSetAuRa', web3);
+    // `6` is to account for possible period of validator set change
+    let stakeWithdrawDisallowPeriod = parseInt(await ValidatorSetContract.instance.methods.stakeWithdrawDisallowPeriod().call());
+    let maxRetriesBlocks = 6 + stakeWithdrawDisallowPeriod;
+
+    let startBlock = await getCurrentBlockNumber(web3);
+    let currentBlock = startBlock;
+    let exc;
+
+    while (currentBlock <= startBlock + maxRetriesBlocks) {
+        currentBlock = await getCurrentBlockNumber(web3);
+        if ( !(await ValidatorSetContract.instance.methods.areStakeAndWithdrawAllowed().call()) ) {
+            console.log(`***** stake/withdraw not allowed now (block ${currentBlock})`);
+            await new Promise(r => setTimeout(r, RETRY_INTERVAL_MS));
+            continue;
+        }
+        try {
+            let tx = await sendTx();
+            return tx;
+        }
+        catch (e) {
+            exc = e;
+            let blocksPassed = (await getCurrentBlockNumber(web3)) - currentBlock;
```

Comment:
It seems that the `blocksPassed` will always be `0` here? Because the `currentBlock` is equal to `await getCurrentBlockNumber(web3)`.

Reply:
`currentBlock` is initially set before the tx is submitted, but this `catch` block is evaluated _after_ tx is executed, so there is at least 1 block difference.
I would expect that it should always be 1 or 2 blocks, but my tests show that sometimes it may even be 5 blocks, I guess because of the slowness of the test.

Reply:
Ok, got it. I have no questions anymore.